py/src/braintrust/logger.py

```diff
@@ -1852,11 +1877,15 @@ def __init__(
             _EXEC_COUNTER += 1
             exec_counter = _EXEC_COUNTER
 
+        type_arg = dict()
+        if type is not None:
+            type_arg = dict(type=type)
```

Comment:
Maybe we should clear out `None` fields in `internal_data` as well in python in `SpanImpl.log`. Seems harmless and lets us avoid extra boilerplate like this.

Reply:
Good idea